drivers/volume/kdmp/kdmp.go

```diff
@@ -231,26 +239,174 @@ func (k *kdmp) UpdateMigratedPersistentVolumeSpec(
+	log.ApplicationRestoreLog(restore).Debugf("started generic restore: %v", restore.Name)
 	volumeInfos := make([]*storkapi.ApplicationRestoreVolumeInfo, 0)
+	for _, bkpvInfo := range volumeBackupInfos {
+		volumeInfo := &storkapi.ApplicationRestoreVolumeInfo{}
+		// wait for pvc to get bound
+		err := wait.ExponentialBackoff(volumeAPICallBackoff, func() (bool, error) {
+			pvc, err := core.Instance().GetPersistentVolumeClaim(bkpvInfo.PersistentVolumeClaim, bkpvInfo.Namespace)
+			if err != nil {
+				return false, nil
+			}
+			if pvc.Spec.VolumeName == "" {
+				return false, nil
+			}
+			volumeInfo.RestoreVolume = pvc.Spec.VolumeName
+			return true, nil
+		})
+		if err != nil {
+			logrus.Errorf("Failed to get volume restore volume: %s, err: %v", bkpvInfo.PersistentVolumeClaim, err)
+			continue
+		}
+		if err != nil {
+			return nil, err
+		}
+		volumeInfo.PersistentVolumeClaim = bkpvInfo.PersistentVolumeClaim
+		volumeInfo.SourceNamespace = bkpvInfo.Namespace
+		volumeInfo.SourceVolume = bkpvInfo.Volume
+		volumeInfo.DriverName = driverName
+
+		// create VolumeBackup CR
+		volBackup := &kdmpapi.VolumeBackup{}
+		volBackup.Annotations = make(map[string]string)
+		volBackup.Annotations[skipResourceAnnotation] = "true"
+		volBackup.Name = getGenericCRName(prefixRestore+"-"+restore.Name, bkpvInfo.Namespace, bkpvInfo.PersistentVolumeClaim)
+		volBackup.Namespace = bkpvInfo.Namespace
+		volBackup.Spec.BackupLocation = kdmpapi.DataExportObjectReference{
+			Kind:       reflect.TypeOf(storkapi.BackupLocation{}).Name(),
+			Name:       restore.Spec.BackupLocation,
+			Namespace:  restore.Namespace,
+			APIVersion: StorkAPIVersion,
+		}
+		volBackup.Spec.Repository = fmt.Sprintf("%s/%s-%s/", prefixRepo, bkpvInfo.Namespace, bkpvInfo.PersistentVolumeClaim)
+		volBackup.Status.SnapshotID = bkpvInfo.BackupID
+		if _, err := kdmpShedOps.Instance().CreateVolumeBackup(volBackup); err != nil {
+			logrus.Errorf("unable to create backup cr: %v", err)
+			return nil, err
+		}
+
+		// create kdmp cr
+		dataExport := &kdmpapi.DataExport{}
+		dataExport.Annotations = make(map[string]string)
+		dataExport.Annotations[skipResourceAnnotation] = "true"
+		dataExport.Name = getGenericCRName(restore.Name, bkpvInfo.Namespace, bkpvInfo.PersistentVolumeClaim)
+		dataExport.Namespace = bkpvInfo.Namespace
+		dataExport.Status.TransferID = volBackup.Namespace + "/" + volBackup.Name
+		dataExport.Spec.Type = kdmpapi.DataExportKopia
+		dataExport.Spec.Source = kdmpapi.DataExportObjectReference{
+			Kind:       reflect.TypeOf(kdmpapi.VolumeBackup{}).Name(),
+			Name:       volBackup.Name,
+			Namespace:  volBackup.Namespace,
+			APIVersion: KdmpAPIVersion,
+		}
+		dataExport.Spec.Destination = kdmpapi.DataExportObjectReference{
+			Kind:       PVCKind,
+			Name:       bkpvInfo.PersistentVolumeClaim,
+			Namespace:  bkpvInfo.Namespace,
+			APIVersion: "v1",
+		}
+		if _, err := kdmpShedOps.Instance().CreateDataExport(dataExport); err != nil {
```

Comment:
Should we check for already exist error and ignore the error, when any retry happens for a volume.

Reply:
applicationcontroller will only retry restore if driver send `Busy` error backup to controller